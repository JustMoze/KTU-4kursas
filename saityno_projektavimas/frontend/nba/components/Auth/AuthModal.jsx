import React, { useState, useEffect } from 'react'
import Button from '@material-ui/core/Button'
import Modal from '@material-ui/core/Modal'
import Input from './Input'
import styled from 'styled-components'
import { makeStyles } from '@material-ui/core/styles'
import { RiLoginCircleLine } from 'react-icons/ri'
import { AiFillCloseCircle } from 'react-icons/ai'
import { NBA_blue } from '../../styles/globalStyle.module.scss'
import { login, register } from '../../utils/Inputs'

const useStyles = makeStyles((theme) => ({
  button: {
    margin: theme.spacing(1),
    width: '70%',
    borderRadius: 30,
    backgroundColor: NBA_blue,
    '&:hover, &:focus': {
      width: '80%',
      backgroundColor: 'green',
      minHeight: 40,
    },
  },
}))

// const InputContainer = styled.div`
//   width: 100%;
//   padding-right: 20px;
//   padding-left: 20px;
// `

const ModalContainer = styled.div`
  width: 100%;
  height: 100%;
  justify-content: center;
  align-items: center;
  background-color: white;
  position: relative;
  border-radius: 10px;
`
const CenterContainer = styled.div`
  padding-top: 30px;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  width: 90%;
`
const CustomModal = styled(Modal)`
  position: absolute;
  width: 40vw;
  height: 60vh;
  min-height: 500px;
  zindex: 10;
  padding: 10px;
  border-radius: 10px;
  justify-content: center;
  align-items: center;
`
const IconContainer = styled.div`
  position: absolute;
  top: 5px;
  cursor: pointer;
  right: 5px;
  min-width: 25px;
  min-height: 25px;
`

const AuthModal = ({ open, handleClose, type }) => {
  const classes = useStyles()
  const [allowSubmit, setAllowSubmit] = useState(false)
  const inputs = type.toLowerCase() === 'register' ? register : login
  const [user, setUser] = useState({})

  const ValidateInput = (input) => {
    return false
  }

  const HandleUserCredentialChange = (e) => {
    let { value, name } = e.target
    setUser({ ...user, [name]: value })
  }

  return (
    <CustomModal
      open={open}
      onBackdropClick={handleClose}
      onEscapeKeyDown={handleClose}
      onClose={handleClose}
      style={{
        top: type.toLowerCase() === 'register' ? '10vh' : '25vh',
        left: '30vw',
        minHeight: type.toLowerCase() === 'register' ? '80vh' : 400,
      }}
    >
      <ModalContainer>
        <IconContainer onClick={handleClose}>
          <AiFillCloseCircle size={25} />
        </IconContainer>
        <CenterContainer>
          {inputs.map((input, index) => {
            return (
              <Input
                key={index}
                error={ValidateInput(input)}
                message="Need to fill it"
                name={input.name}
                value={user[`${input.name}`]}
                handleChange={(e) => HandleUserCredentialChange(e)}
                type={input.type}
              />
            )
          })}
          <div
            style={{
              width: '100%',
              marginTop: 20,
              display: 'flex',
              justifyContent: 'center',
            }}
          >
            <Button
              variant="contained"
              size="small"
              className={classes.button}
              disabled={!allowSubmit}
              startIcon={<RiLoginCircleLine size={25} />}
            >
              Save
            </Button>
          </div>
        </CenterContainer>
      </ModalContainer>
    </CustomModal>
  )
}

export default AuthModal
