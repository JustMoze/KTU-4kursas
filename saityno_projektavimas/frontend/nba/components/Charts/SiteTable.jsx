import React, { useState, useEffect } from 'react'
import { Thead, Tbody, Tr, Th } from 'react-super-responsive-table'
import 'react-super-responsive-table/dist/SuperResponsiveTableStyle.css'
import styled from 'styled-components'
import { NBA_red, NBA_white } from '../../styles/globalStyle.module.scss'
import { TiChevronRight, TiChevronLeft } from 'react-icons/ti'
import Loader from '../Loader/Loader'
import config from '../../config.json'
import AppSnackbar from './../Snackbar/Snackbar'
import {
  CustomTd,
  CustomTable,
  CustomTr,
  CasualTh,
  CustomTdPlayer,
} from './../../utils/Chart'

const PlayerRow = styled(Tr)`
  position: relative;
  cursor: pointer;
  background-color: ${(props) =>
    props.price > 9000000 ? '#B9BEC9' : '#ffffff'};
  &:hover {
    background-color: ${(props) => props.color};
    color: ${NBA_white};
  }
`
const ArrowContainer = styled.div`
  width: 45px;
  height: 45px;
  cursor: pointer;
  margin-left: 15px;
  border-radius: 50%;
  justifycontent: center;
  align-items: center;
  background-color: #b9bec9;
  &:hover {
    background-color: ${(props) => (props.enable ? props.color : '#b9bec9')};
    color: ${NBA_white};
  }
`
const CustomBody = styled(Tbody)`
  margin-top: 25px;
`
function SiteTable({ color, abbreviation, handleClick }) {
  const [loading, setLoading] = useState(true)
  const [chartPlayers, setChartPlayers] = useState()
  const [number, setNumber] = useState(0)
  const [players, setPlayers] = useState()
  const [allowLeft, setAllowLeft] = useState(true)
  const [allowRight, setAllowRight] = useState(true)
  const [openSnackbar, setSnackbarOpen] = useState(false)
  const [errMessage, setErrMessage] = useState('')
  useEffect(() => {
    fetch(`${config.API}players/team/${abbreviation}`)
      .then((response) => response.json())
      .then((data) => {
        setPlayers(data)
        setChartPlayers(data.slice(0, 10))
        setLoading(false)
      })
      .catch((ex) => {
        setSnackbarOpen(true)
        setErrMessage(ex.message)
        setLoading(false)
      })
  }, [abbreviation])

  const handleClose = (event, reason) => {
    if (reason === 'clickaway') {
      setSnackbarOpen(false)
    }
    setSnackbarOpen(false)
  }
  const handleArrowClick = (side) => {
    switch (side) {
      case 'left':
        let newNumber = number - 1
        if (newNumber > -1) {
          setNumber(newNumber)
          setChartPlayers(players.slice(newNumber * 10, newNumber * 10 + 9))
          if (allowRight === false) {
            setAllowRight(true)
          }
          if (newNumber === 0) {
            setAllowLeft(false)
          }
        } else {
          setAllowLeft(false)
        }
        break
      case 'right':
        let newNumberRight = number + 1
        if (newNumberRight < players.length / 10) {
          if (newNumberRight * 10 + 9 > players.length) {
            setChartPlayers(players.slice(newNumberRight * 10, players.length))
            setNumber(newNumberRight)
            setAllowLeft(true)
            setAllowRight(false)
          } else {
            setChartPlayers(players.slice(newNumberRight * 10, players.length))
            setNumber(newNumberRight)
            setAllowLeft(true)
          }
        } else if (
          newNumberRight ===
            (players.length - findResidue(players.length)) / 10 &&
          findResidue(players.length) > 0
        ) {
          setNumber(newNumberRight)
          setAllowRight(false)
          setChartPlayers(
            players.slice(
              players.length - findResidue(players.length),
              players.length
            )
          )
        }
      default:
        break
    }
  }
  const findResidue = (number) => {
    return number % 10
  }
  console.log('number', number)
  return (
    <>
      {loading ? (
        <div
          style={{
            width: '100%',
            height: '100vh',
            justifyContent: 'center',
            alignItems: 'center',
          }}
        >
          <Loader size={40} />
        </div>
      ) : (
        <>
          {openSnackbar ? (
            <AppSnackbar
              message={errMessage}
              open={openSnackbar}
              handleClose={handleClose}
              severity="error"
            />
          ) : (
            <>
              <CustomTable>
                <Thead>
                  <CustomTr color={color}>
                    <Th>Player</Th>
                    <CasualTh>Position</CasualTh>
                    <CasualTh>Number</CasualTh>
                  </CustomTr>
                </Thead>
                <CustomBody>
                  {chartPlayers.map((player) => {
                    return (
                      <PlayerRow
                        key={player._id}
                        price={player.price}
                        color={color}
                        onClick={() => handleClick(player._id)}
                      >
                        <CustomTdPlayer>{player.fullName}</CustomTdPlayer>
                        <CustomTd>{player.position}</CustomTd>
                        <CustomTd>{player.number}</CustomTd>
                      </PlayerRow>
                    )
                  })}
                </CustomBody>
              </CustomTable>
              <div
                style={{
                  position: 'relative',
                  justifyContent: 'flex-end',
                  alignItems: 'center',
                  display: 'flex',
                  float: 'right',
                  marginTop: '1%',
                  marginRight: '3.3%',
                  width: '50%',
                  height: 50,
                }}
              >
                <ArrowContainer
                  color={color}
                  enable={number > 0 ? true : false}
                  onClick={() => handleArrowClick('left')}
                >
                  {number !== 0 && <TiChevronLeft size={45} color="#ffffff" />}
                </ArrowContainer>
                <ArrowContainer
                  color={color}
                  enable={allowRight}
                  onClick={() => handleArrowClick('right')}
                >
                  {allowRight && <TiChevronRight size={45} color="#ffffff" />}
                </ArrowContainer>
              </div>
            </>
          )}
        </>
      )}
    </>
  )
}

export default SiteTable
