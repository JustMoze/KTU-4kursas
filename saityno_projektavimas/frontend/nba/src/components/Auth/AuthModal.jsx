import React from 'react'
import Button from '@material-ui/core/Button'
import Modal from '@material-ui/core/Modal'
import Input from './Input'
import styled from 'styled-components'
import { makeStyles } from '@material-ui/core/styles'
import { RiLoginCircleLine } from 'react-icons/ri'
import { AiFillCloseCircle } from 'react-icons/ai'
import { login, register, validateLength } from '../../../utils/Inputs'
import has from 'lodash/has'
import { validateEmail } from './../../../utils/Inputs'

const useStyles = makeStyles((theme) => ({
  button: {
    margin: theme.spacing(1),
    width: '70%',
    borderRadius: 30,
    backgroundColor: '#1e438a',
    '&:hover, &:focus': {
      width: '80%',
      backgroundColor: 'green',
      minHeight: 40,
    },
  },
}))
export const LogoContainer = styled.div`
  background-color: ${(props) => props.color};
  width: 30%;
  height: 60px;
  justify-content: center;
  align-self: center;
  align-items: center;
  display: flex;
  margin-bottom: 25px;
  padding: 3px;
  border-radius: 15px;
`
const ModalContainer = styled.div`
  width: 100%;
  height: 100%;
  justify-content: center;
  align-items: center;
  background-color: white;
  position: relative;
  border-radius: 10px;
`
const CenterContainer = styled.div`
  padding-top: 30px;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  width: 90%;
`
const CustomModal = styled(Modal)`
  position: absolute;
  background-color: #d4d7dd;
  width: 40vw;
  height: 60vh;
  min-height: 500px;
  zindex: 10;
  border-radius: 10px;
  justify-content: center;
  align-items: center;
`
const IconContainer = styled.div`
  position: absolute;
  top: 5px;
  cursor: pointer;
  right: 5px;
  min-width: 25px;
  min-height: 25px;
`

const AuthModal = ({
  open,
  handleClose,
  type,
  user,
  handleChange,
  handleSubmit,
  color,
}) => {
  const classes = useStyles()
  const inputs = type.toLowerCase() === 'register' ? register : login
  var errArr = {}

  const ValidateInput = (input) => {
    let { validation } = input
    if (has(validation, 'email')) {
      if (!validateEmail(user[`${input.name}`])) {
        errArr[`${input.name}`] = 1
        return 'Email does not match format'
      } else {
        errArr[`${input.name}`] = 0
      }
    } else if (has(validation, 'min') && has(validation, 'max')) {
      let { min, max } = validation
      let errArrs = validateLength(min, max, user[`${input.name}`], input.name)
      if (errArrs.length > 0) {
        errArr[`${input.name}`] = 1
        return errArrs[0]
      } else {
        errArr[`${input.name}`] = 0
      }
    }
    if (errArr[`${input.name}`] === 0) {
      return ''
    }
  }
  const AllowSubmit = () => {
    let counter = 0
    for (let prop in errArr) {
      if (errArr[prop] === 0) {
        counter++
      }
    }
    if (counter === inputs.length) {
      return true
    } else return false
  }
  return (
    <CustomModal
      open={open}
      onBackdropClick={handleClose}
      onEscapeKeyDown={handleClose}
      onClose={handleClose}
      style={{
        top: type.toLowerCase() === 'register' ? '10vh' : '25vh',
        left: '30vw',
        minHeight: type.toLowerCase() === 'register' ? '80vh' : 400,
      }}
    >
      <ModalContainer>
        <IconContainer id="iconContainer" onClick={handleClose}>
          <AiFillCloseCircle size={25} />
        </IconContainer>
        <CenterContainer>
          <div
            style={{
              width: '100%',
              display: 'flex',
              alignSelf: 'center',
              justifyContent: 'center',
            }}
          >
            <LogoContainer color={color}>
              <img
                src="https://ak-static.cms.nba.com/wp-content/uploads/logos/leagues/logo-nba.svg"
                height="100%"
              />
            </LogoContainer>
          </div>
          {inputs.map((input, index) => {
            return (
              <Input
                color={color}
                key={index}
                error={ValidateInput(input) === '' ? false : true}
                message={ValidateInput(input)}
                name={`${type}_${input.name}`}
                value={user[`${input.name}`]}
                handleChange={(e) => handleChange(e)}
                type={input.type}
              />
            )
          })}
          <div
            style={{
              width: '100%',
              marginTop: 20,
              display: 'flex',
              justifyContent: 'center',
            }}
          >
            <Button
              variant="contained"
              size="small"
              style={{ backgroundColor: color, color: '#ffffff' }}
              className={classes.button}
              onClick={() => handleSubmit(type)}
              disabled={!AllowSubmit()}
              startIcon={<RiLoginCircleLine size={25} />}
            >
              Save
            </Button>
          </div>
        </CenterContainer>
      </ModalContainer>
    </CustomModal>
  )
}

export default AuthModal
