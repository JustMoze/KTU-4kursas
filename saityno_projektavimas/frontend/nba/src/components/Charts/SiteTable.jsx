import React, { useState, useEffect } from 'react'
import { Thead, Tbody, Tr, Th } from 'react-super-responsive-table'
import styled from 'styled-components'
import { TiChevronRight, TiChevronLeft } from 'react-icons/ti'
import Loader from '../Loader/Loader'
import config from '../../../config.json'
import AppSnackbar from './../Snackbar/Snackbar'
import {
  CustomTd,
  CustomTable,
  CustomTr,
  CasualTh,
  CustomTdPlayer,
} from './../../../utils/Chart'

export const ArrowsRow = styled(Tr)`
    display: flex;
    position: relative;
    width: 100%;
    height: 0px;
    justify-content: flex-end;
    @media (max-width: 768px) {
      background-color: ${props => props.color};
    }
`;

export const PlayerRow = styled(Tr)`
  background-color: ${(props) =>
    props.price > 9000000 ? '#B9BEC9' : '#ffffff'};
  position: relative;
  cursor: pointer;
  &:hover {
    background-color: ${(props) => props.color};
    color: #ffffff;
  }
`
export const ArrowContainer = styled.div`
  width: 45px;
  height: 45px;
  cursor: pointer;
  margin-left: 15px;
  border-radius: 50%;
  justifycontent: center;
  align-items: center;
  background-color: #b9bec9;
  &:hover {
    background-color: ${(props) => (props.enable ? props.color : '#b9bec9')};
    color: #ffffff;
  }
`
const CustomBody = styled(Tbody)`
  margin-top: 25px;
`
const NavigationContainer = styled.div`
    position: relative;
    display: flex;
    
    justify-content: flex-end;
    align-items: center;
    float: right;
    margin-top: 1%;
    margin-right: 3.3%;
    width: 50%;
    height: 50px;
    z-index: 5;
    @media (max-width: 768px) {
      margin-right: 0%;
      margin-top: 10px;
    }
`;
function SiteTable({ color, abbreviation, handleClick, center = false }) {
  const [loading, setLoading] = useState(true)
  const [chartPlayers, setChartPlayers] = useState()
  const [number, setNumber] = useState(0)
  const [players, setPlayers] = useState()
  const [allowLeft, setAllowLeft] = useState(true)
  const [allowRight, setAllowRight] = useState(true)
  const [openSnackbar, setSnackbarOpen] = useState(false)
  const [errMessage, setErrMessage] = useState('')
  useEffect(() => {
    fetch(`${config.API}players/team/${abbreviation}`)
      .then((response) => response.json())
      .then((data) => {    
        setPlayers(data)
        setChartPlayers(data.slice(0, 11))
        setLoading(false)
      })
      .catch((ex) => {
        setSnackbarOpen(true)
        setErrMessage(ex.message)
        setLoading(false)
      })
  }, [abbreviation])

  const handleClose = (event, reason) => {
    if (reason === 'clickaway') {
      setSnackbarOpen(false)
    }
    setSnackbarOpen(false)
  }
  const handleArrowClick = (side) => {
    switch (side) {
      case 'left':
        let newNumber = number - 1
        if (newNumber > -1) {
          setNumber(newNumber)
          setChartPlayers(players.slice(newNumber * 10, newNumber * 10 + 9))
          if (allowRight === false) {
            setAllowRight(true)
          }
          if (newNumber === 0) {
            setAllowLeft(false)
          }
        } else {
          setAllowLeft(false)
        }
        break
      case 'right':
        let newNumberRight = number + 1
        if (newNumberRight < players.length / 10) {
          if (newNumberRight * 10 + 9 > players.length) {
            setChartPlayers(players.slice(newNumberRight * 10, players.length))
            setNumber(newNumberRight)
            setAllowLeft(true)
            setAllowRight(false)
          } else {
            setChartPlayers(players.slice(newNumberRight * 10, players.length))
            setNumber(newNumberRight)
            setAllowLeft(true)
          }
        } else if (
          newNumberRight ===
            (players.length - findResidue(players.length)) / 10 &&
          findResidue(players.length) > 0
        ) {
          setNumber(newNumberRight)
          setAllowRight(false)
          setChartPlayers(
            players.slice(
              players.length - findResidue(players.length),
              players.length
            )
          )
        }
      default:
        break
    }
  }
  const findResidue = (number) => {
    return number % 10
  }
  return (
    <>
      {loading ? (
        <div
          style={{
            width: '100%',
            height: '100vh',
            justifyContent: 'center',
            alignItems: 'center',
          }}
        >
          <Loader size={40} />
        </div>
      ) : (
        <>
          {openSnackbar ? (
            <AppSnackbar
              message={errMessage}
              open={openSnackbar}
              handleClose={handleClose}
              severity="error"
            />
          ) : (
            <>
              <CustomTable>
                <Thead>
                  <CustomTr color={color}>
                    <Th>Player</Th>
                    <CasualTh>Position</CasualTh>
                    <CasualTh>Number</CasualTh>
                  </CustomTr>
                </Thead>
                <CustomBody>
                  {chartPlayers.map((player) => {
                    return (
                      <PlayerRow
                        key={player._id}
                        price={player.price}
                        color={color}
                        onClick={() => handleClick(player._id)}
                      >
                        <CustomTdPlayer>{player.fullName}</CustomTdPlayer>
                        <CustomTd>{player.position}</CustomTd>
                        <CustomTd>{player.number}</CustomTd>
                      </PlayerRow>
                    )
                  })}
                  <ArrowsRow color={color}>
                    <NavigationContainer style={{width: center ? '100%' : '50%', justifyContent: center ? "center" : "flex-end"}}>
                      <ArrowContainer
                        color={color}
                        enable={number > 0 ? true : false}
                        onClick={() => handleArrowClick('left')}
                      >
                        {number !== 0 && (
                          <TiChevronLeft size={45} color="#ffffff" />
                        )}
                      </ArrowContainer>
                      <ArrowContainer
                        color={color}
                        enable={allowRight}
                        onClick={() => handleArrowClick('right')}
                      >
                        {allowRight && (
                          <TiChevronRight size={45} color="#ffffff" />
                        )}
                      </ArrowContainer>
                    </NavigationContainer>
                  </ArrowsRow>
                </CustomBody>
              </CustomTable>
            </>
          )}
        </>
      )}
    </>
  )
}

export default SiteTable
