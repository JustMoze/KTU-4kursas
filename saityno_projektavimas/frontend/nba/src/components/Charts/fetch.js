export function myFetch(){
    return fetch('https://nba-modestas.herokuapp.com/players/team/UTA').then((response) => response.json())
}