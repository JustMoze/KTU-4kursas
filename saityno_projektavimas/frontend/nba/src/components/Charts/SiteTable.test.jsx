import React from 'react'
import { shallow, mount } from 'enzyme'
import SiteTable, { ArrowContainer, ArrowsRow, PlayerRow } from './SiteTable'
import toJson from 'enzyme-to-json'
import renderer from 'react-test-renderer'
import {myFetch} from './fetch'
import 'jest-styled-components'
import AuthModal from '../Auth/AuthModal'

jest.mock('./fetch', () => {
  return {
    myFetch: jest.fn(),
  }
})

// describe('Fetching', () => {
//   afterEach(() => {
//     jest.clearAllMocks()
//   })
//   it('should fetch data correctly', async () => {
//     const mResponse = {
//       data: {
//         _id: '5f62427b92c3b8345482f4b2',
//         fullName: 'OG Anunoby',
//         position: 'F',
//         number: '#3',
//         weight: '232 lbs',
//         height: '6 ft 7in',
//         age: '23 years',
//         mpg: 23.3,
//         fg: 48,
//         threePt: 36.5,
//         ft: 65,
//         ppg: 7.8,
//         rpg: 3.5,
//         bpg: 0.4,
//         apg: 1,
//         team: 'TOR',
//         _v: 0,
//         price: 4318056,
//         foto:
//           'https://ak-static.cms.nba.com/wp-content/uploads/headshots/nba/latest/260x190/1628384.png',
//         videoId: 't0aLmBCXoqA',
//       },
//     }
//     const handleClick = jest.fn();
//     const initProps = {
//         color: '#ffffff',
//         abbreviation: 'UTA',
//         handleClick: handleClick,
//     };
//     const mProps = { id: 1 }
//     myFetch.mockResolvedValueOnce(mResponse)
//     const wrapper = mount(<AuthModal {...initProps}></AuthModal>)
//     expect(wrapper.exists).toBeTruthy()
//     await act(async () => {
//       await new Promise((resolve) => setTimeout(resolve, 0))
//     })
//     wrapper.update()
//     expect(wrapper.find('PlayerRow').length).toBeGreaterThanOrEqual(1);
//     expect(myFetch).toBeCalledWith(1)
//   })
// });

describe('SiteTable component', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  const props = {
    color: '#ffffff',
    abbreviation: 'UTA',
    handleClick: jest.fn(),
  }
  const price = 9000001
  it('Renders footer correctly', () => {
    let wrapper = shallow(<SiteTable {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  test('ArrowContainer works', () => {
    const arrContainer = renderer
      .create(<ArrowContainer color={props.color} enable={true} />)
      .toJSON()
    expect(arrContainer).toHaveStyleRule('background-color', props.color, {
      modifier: ':hover',
    })
  })
  test('PlayerRow works', () => {
    const playerRow = renderer
      .create(<PlayerRow color={props.color} price={price} />)
      .toJSON()
    expect(playerRow).toHaveStyleRule('background-color', props.color, {
      modifier: ':hover',
    })
  })
  test('ArrowsRow works', () => {
    const arrContainer = renderer
      .create(<ArrowsRow color={props.color} />)
      .toJSON()
    expect(arrContainer).toHaveStyleRule('background-color', props.color, {
      media: '(max-width: 768px)',
    })
  })

//   it('map function checking', () => {
//     expect(wrapper.find(PlayerRow)).to.have.length()
//     expect(wrapper.find('tbody').children().find('tr')).to.have.length(
//       cats.length
//     )
//   })
})
