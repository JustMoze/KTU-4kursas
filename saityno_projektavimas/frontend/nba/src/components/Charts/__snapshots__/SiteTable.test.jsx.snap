// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteTable component Renders footer correctly 1`] = `
<Fragment>
  <div
    style={
      Object {
        "alignItems": "center",
        "height": "100vh",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Loader
      size={40}
    />
  </div>
</Fragment>
`;
